adoc-mode.el

```diff
@@ -44,6 +44,15 @@
 
 (require 'cl-lib)
 (require 'tempo)
+(require 'mouse)
+(require 'image)
+(eval-when-compile
+  (when (version< emacs-version "29.0") ;; when-let has moved from subr-x to subr
+    (require 'subr-x)
+    ))
+
+(declare-function imagep "image.c")
```

Comment:
If you're using these declares I guess you don't need to require `image`. I'm not sure if requiring `mouse` is needed as well, if you did it just to see if it would affected the CI tests.